File: apple/Modules/Networking/Sources/DavsClient/DavsClient.swift
//
//  DavsClient.swift
//  
//
//  Created by Kamaal M Farah on 11/06/2024.
//

import Foundation

public actor DavsClient {
    public let health: DavsHealthClient
    public let users: DavsUsersClient
    public let contacts: DavsContactsClient

    init() {
        let secrets = SecretsJSON()
        let secretsContent = secrets.content
        let baseURL = secretsContent.davsBaseURL.appending(path: "api/v1")
        self.health = DavsHealthClient(baseURL: baseURL)
        self.users = DavsUsersClient(baseURL: baseURL, apiKey: secretsContent.davsAPIKey)
        self.contacts = DavsContactsClient(baseURL: baseURL)
    }

    public static let shared = DavsClient()

    public func setAuthorizationToken(_ token: String) async {
        await DavsClientState.shared.setAuthorizationToken(token)
    }

    public func clearAuthorizationToken() async {
        await DavsClientState.shared.clearAuthorizationToken()
    }
}

File: web/src/server/clients/davs-client/davs-client.ts
import settings from './settings';
import DavsUsersClient from './users';

const { DAVS_API_KEY, DAVS_SERVER_BASE_URL } = settings;

class DavsClient {
  users: DavsUsersClient;

  constructor() {
    const baseURL = new URL(`${DAVS_SERVER_BASE_URL}/api/v1`);
    this.users = new DavsUsersClient({
      davsBaseURL: baseURL,
      apiKey: DAVS_API_KEY,
    });
  }
}

const davsClient = new DavsClient();

export default davsClient;
